R/get_cded_dem.R

```diff
@@ -19,33 +20,70 @@
 #'
 #' @examples
 #' \dontrun{
-#' get_cded_dem(aoi_bb = file.path(fid$shape_dir_1010[2],"aoi_snapped.gpkg"),
-#'     res = 5, out_dir = fid$cov_dir_1020[2]))
+#' get_cded_dem(
+#'   aoi = fs::path(PEMprepr::read_fid()$dir_1020_covariates$path_abs, "25m", "template.tif"),
+#'   res = FALSE,
+#'   out_dir = PEMprepr::read_fid()$dir_1020_covariates$path_abs,
+#'   write_output = TRUE
+#'   overwrite = FALSE)
 #' }
 get_cded_dem <- function(aoi = fs::path(PEMprepr::read_fid()$dir_1020_covariates$path_abs, "25m", "template.tif"),
-                         res = 5,
+                         res = NULL,
                          out_dir = PEMprepr::read_fid()$dir_1020_covariates$path_abs,
                          write_output = TRUE,
                          overwrite = FALSE, ...) {
+
   if (inherits(aoi, c("character"))) {
     aoi <- terra::rast(aoi)
   } else if (!inherits(aoi, c("SpatRaster"))) {
-    cli::cli_abort("{.var aoi} must be an spatRast or a path to a .tif file")
+    cli::cli_abort("{.var aoi} must be a SpatRaster or a path to a file")
   }
 
-  if (!is.numeric(res)) {
-    cli::cli_abort("{.var res} must be numeric")
-  }
+  aoi_res <- terra::res(aoi)[1]
+
+  if (!is.null(res)) { 
+    if (!is.numeric(res)) {
+      cli::cli_abort("{.var res} must be numeric")
+    }
 
-  output_dir <- fs::path(out_dir, paste0(res, "m"))
+    if(aoi_res != res) {
 
-  cded_raw <- bcmaps::cded_terra(aoi, ...)
+      cli::cli_alert_warning(
+        "Updating cded to resolution specified"
+      )
+
+      aoi_out <- aoi
+      terra::res(aoi_out) <- res
+      aoi_template <- terra::resample(aoi, aoi_out)
+
+    } else {
+
+      aoi_template <- aoi
+      cli::cli_alert_warning(
+        "Specified res matches template resolution, output cded resolution will be {.var res}"
+      )
+    }
 
-  cded <- terra::project(cded_raw, aoi)
+    output_res <- res
 
+  } else {
+
+    aoi_template <- aoi
+    output_res <- aoi_res
+    cli::cli_alert_warning(
+      "No spatial resolution specified, cded output spatial resolution will match input raster"
+    )
+  }
+
+  cded_raw <- bcmaps::cded_terra(aoi_template, ...)
+  #cded_raw <- bcmaps::cded_terra(aoi_template)#, ...)
+
+  cded <- terra::project(cded_raw, aoi_template)
 
-  ## check if folder exists
   if (write_output) {
+
+    output_dir <- fs::path(out_dir, paste0(output_res, "m"))
+
     if (!fs::dir_exists(fs::path(output_dir))) {
       fs::dir_create(fs::path(output_dir), recurse = TRUE)
     }
```

Comment:
```suggestion
    fs::dir_create(fs::path(output_dir), recurse = TRUE)
```

`dir_create()` silently fails if dir already exists, so don't need to check its existence